### PR TITLE
Add TKR API v1alpha3 types

### LINKED ISSUE #878: TKR v1alpha3 API: types
## TanzuKubernetesRelease
The TKR resource represents a Tanzu Kubernetes release. TKR is a “summary” object: it aggregates OSImage and other objects shipped together in a single release. It also defines metadata applicable to the release as a whole. 

Several Tanzu Kubernetes releases MAY ship exactly the same Kubernetes version, but different OSImages (or other objects — see below), for example, to provide fixes to OS components. 

```yaml
apiVersion: run.tanzu.vmware.com/v1alpha3
kind: TanzuKubernetesRelease

metadata:
  name: v1.21.2---vmware.1-tkg.1
  labels:
    corp.example.com/hardened: "true" # (an example label)

spec:
  # version of this Tanzu Kubernetes release. 
  # MUST conform to SemVer and be prefixed with 'v'.
  version: v1.21.2+vmware.1-tkg.1

  bootstrapPackages:
  - name: kapp-controller.carvel.dev.0.30.0
  - name: tkr-addons-repo.run.tanzu.1.21.2---vmware.1-tkg.1

  kubernetes:
    # version is the build version of Kubernetes, which is the basis of the
    # Tanzu Kubernetes release. MUST conform to SemVer and be prefixed with 'v'.
    version: v1.21.2+vmware.1
    imageRepository: # defaults to `projects.registry.vmware.com/tkg`
    etcd:
      imageRepository: # defaults to `projects.registry.vmware.com/tkg`
      imageTag: v3.5.0
    pause:
      imageRepository: # defaults to `projects.registry.vmware.com/tkg`
      imageTag: "3.4.1"
    coredns:
      imageRepository: # defaults to `projects.registry.vmware.com/tkg`
      imageTag: v1.8.0_vmware.5

status:
  conditions:
  - lastTransitionTime: "2021-10-26T18:43:24Z"
    status: "True"
    type: Compatible
  - lastTransitionTime: "2021-10-26T18:43:18Z"
    status: "True"
    type: UpdatesAvailable
    message: '[v1.21.2+vmware.1-tkg.2]'
```

Status conditions are set by the TKR controller based on the observed state of the management cluster, e.g. presence of more recent TKRs usable as potential updates. 

An Administrator / Platform Operator can deactivate a TKR by adding the deactivated: "" label to the object: objects with this label will be excluded from lookups. Users are free to use the value of the label to indicate the “reason for deactivation” (in compliance with Kubernetes label value syntax, of course). In a CLI command to mark a TKR as deactivated, we could take --reason as an optional arg.

## OSImage
The OSImage resource represents a single OS image unit addressable in the relevant infrastructure provider. The type of the image (corresponding to the infrastructure provider) is specified in the .spec.image.type field.

For example, for AWS, every AMI is represented by an OSImage object with spec.image.type: ami. 
An AMI exists for every <k8s-version, os-name, os-arch, os-version> combination in every Availability Zone.

```yaml
apiVersion: run.tanzu.vmware.com/v1alpha3
kind: OSImage

metadata:
  name: v1.21.2---vmware.1-tkg.1-ami-0f2e5eec7ae0a1986
  labels:
    run.tanzu.vmware.com/tkr: v1.21.2---vmware.1-tkg.1
    run.tanzu.vmware.com/kubernetesVersion: v1.21.2---vmware.1

spec:
  # kubernetesVersion is the version of Kubernetes components installed in the OS image.
  kubernetesVersion: v1.21.2+vmware.1

  # os describes the OS of the OS image.
  os:
    # type of the OS kernel ("linux" or "windows")
    type: linux
    # name of the OS
    name: ubuntu
    # arch is the CPU architecture of the OS
    arch: amd64
    # version of the OS
    version: "20.04"

  # image contains (provider specific) image details necessary to use it
  image:
    # type of the image ("ova", "ami", "azure", "vmop")
    type: ami
    ref:
      id: ami-0f2e5eec7ae0a1986
      partition: aws
      region: ap-northeast-1

status:
  conditions:
  - lastTransitionTime: "2021-10-26T18:43:24Z"
    status: "True"
    type: Compatible
  - lastTransitionTime: "2021-10-26T18:43:18Z"
    status: "True"
    type: UpdatesAvailable
    message: '[v1.21.2+vmware.1-tkg.2]'
```

Status conditions are set by the TKR controller based on the observed state of the management cluster, e.g. presence of OSImages shipped with more recent TKRs usable as potential updates. 

To enable discoverability, OSImage objects should be additionally labeled by the Labeling Controller. OSImage .spec fields are converted to labels the following way:
* .os fields are prefixed with os- to create labels:
  * os-type
  * os-name
  * os-arch
  * os-version
* .image.type field creates label:
  * image-type
* .image.ref fields are prefixed with the value of .image.type to create labels, for example (for the above OSImage):
  * ami-id
  * ami-partition
  * ami-region

An Administrator / Platform Operator can deactivate an OSImage by adding the deactivated: "" label to the object: objects with this label will be excluded from lookups.

### PR DESCRIPTION
**Pull request by imikushin**
### What this PR does / why we need it

This adds Go types for TKR API v1alpha3.

### Which issue(s) this PR fixes

Fixes #878. 

### Describe testing done for PR

No tests yet: just adding Go type definitions (structs and fields).

### Release note

```release-note

```

### PR Checklist

- [X] Squash the commits into one or a small number of logical commits
      
- [X] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [X] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: apis/run/v1alpha3/osimage_types.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// OSInfo describes the "OS" part of the OSImage, defined by the Operating System's name, version and CPU architecture.
+type OSInfo struct {
+	Name    string `yaml:"name"`
```
Comment (prkalle) [inline apis/run/v1alpha3/osimage_types.go:13]: Question: I see `type` field being mentioned in the design document. Is it intentional to not include `type` field?

**Review by prkalle — COMMENTED** [review]
LGTM. I have one question , otherwise the changes are good. Thanks!

File: apis/run/v1alpha3/osimage_types.go (hunk 1 above)
Comment (imikushin) [reply apis/run/v1alpha3/osimage_types.go:13]: Not intentional, thanks!

Comment (imikushin) [reply apis/run/v1alpha3/osimage_types.go]: Fixed

**Review by prkalle — APPROVED** [review]
LGTM.

**Review by anujc25 — APPROVED** [review]

**Pull request merged**